logging/src/main/java/org/wildfly/swarm/logging/LoggingFraction.java

```diff
@@ -51,6 +60,9 @@
 
     public static final String COLOR_PATTERN = "COLOR_PATTERN";
 
+    public static LoggingFraction loggingFraction;
```

Comment:
I think this is where the hang-up is.

There's two flavors of configuration methods for a given fraction.

The static methods, such as createDebugLoggingFraction() is a helper factory method.  It's a starting point, so you're not starting from scratch with `new LoggingFraction()` and having to do everything by-hand, from nil.

Ultimately, these static methods are doing `new LoggingFraction()`, and then calling a series of *instance* methods on that object to do work.  Ultimately the fraction is returned to the user (or the container), and no static state is or should be held.  Ever.

When we speak of chaining, that means *instance* methods, which return *LoggingFraction* and do work.

Start there.

Once you have friendly instance methods that can be applied to *any* instance of a `LoggingFraction` object, *then* you can think about a handy-dandy helper static factory method just to make things easier.

Bottom line:

* Do work in an instance method
* Have that instance method return `this`
* Once you can work with instance methods, _then_ ponder helper factory methods to help people get started for the 80% case.
* Never hold static state. Ever.  Never.  Ever.  It's a smell if you do.